tuf/testutils/swizzler.go

```diff
@@ -105,17 +104,19 @@ func signedFromStore(cache store.MetadataStore, role string) (*data.Signed, erro
 
 // NewMetadataSwizzler returns a new swizzler when given a gun,
 // mapping of roles to initial metadata bytes, and a cryptoservice
-func NewMetadataSwizzler(gun string, initialMetadata map[string][]byte,
+func NewMetadataSwizzler(gun data.GUN, initialMetadata map[data.RoleName][]byte,
 	cryptoService signed.CryptoService) *MetadataSwizzler {
 
-	var roles []string
-	for roleName := range initialMetadata {
-		roles = append(roles, roleName)
+	var roles []data.RoleName
+	metadata := make(map[data.RoleName][]byte)
```

Comment:
@grantleehoffman The client update test that's failing (and possibly others) assume that as the data in the server data is swizzled, the metadata map is updated, because previously we were just mutating the initial metadata map.  Here a new one is created, so the original is never updated.

2 ways to fix it are to either make the original map mutable, or in the test, iterate over the roles in the swizzler and call `GetSized` on the swizzler's cache instead of iterating over the metadata map.